src/internal/restgateway/restgateway.go

```diff
@@ -0,0 +1,112 @@
+package restgateway
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"google.golang.org/grpc"
+
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"github.com/pachyderm/pachyderm/v2/src/pps"
+
+	"github.com/pachyderm/pachyderm/v2/src/admin"
+	"github.com/pachyderm/pachyderm/v2/src/auth"
+	"github.com/pachyderm/pachyderm/v2/src/debug"
+	"github.com/pachyderm/pachyderm/v2/src/enterprise"
+	"github.com/pachyderm/pachyderm/v2/src/identity"
+	"github.com/pachyderm/pachyderm/v2/src/license"
+	"github.com/pachyderm/pachyderm/v2/src/proxy"
+	"github.com/pachyderm/pachyderm/v2/src/transaction"
+	"github.com/pachyderm/pachyderm/v2/src/version/versionpb"
+	"github.com/pachyderm/pachyderm/v2/src/worker"
+)
+
+func NewMux(ctx context.Context, grpcConn *grpc.ClientConn) http.Handler {
+
+	ctx = pctx.Child(ctx, "restgateway")
+
+	mux := runtime.NewServeMux(runtime.WithIncomingHeaderMatcher(func(s string) (string, bool) {
+		if s != "Content-Length" {
+			return s, true
+		}
+		return s, false
+	}))
+
+	err := pps.RegisterAPIHandler(ctx, mux, grpcConn)
+	if err != nil {
+		log.Error(ctx, "failed to register pps api with grpc-gateway")
+		return nil
```

Comment:
I think we should return these errors rather than logging them.  Otherwise, it's going to be a support escalation when this codepath runs; "I'm trying to use console but it doesn't load".  Then people will look at the error logs from startup and see these errors, but probably not without a ticket being created and a debug dump requested, and someone here manually looking at the debug dump.  Meanwhile, if we return the errors all the way up the chain, then pachd will simply fail to start or upgrade, and that will be less confusing.